services/graphql-server/utils/ast2sdl.go

```diff
@@ -2,72 +2,119 @@ package utils
 
 import (
 	"fmt"
+	"reflect"
 	"strings"
 
 	"github.com/graphql-go/graphql"
 	"github.com/graphql-go/graphql/language/ast"
+	"github.com/sirupsen/logrus"
 )
 
-// ResolveParamsToSDLQuery - generates SDL query from graphql.ResolveParams object
-func ResolveParamsToSDLQuery(queryName string, rp graphql.ResolveParams, args string) string {
-	builder := strings.Builder{}
+// GqlRequestConfig - type to store all data for gql request
+type GqlRequestConfig struct {
+	Query          string
+	VariableValues map[string]interface{}
+}
+
+type sdlQuery struct {
+	builder        *strings.Builder
+	fragmentsToAdd []string
+	addedFragments map[string]bool
+	variables      map[string]string
```

Comment:
I thought `variables` was key-value but it looks like it's key-type? I'm a bit confused

Reply:
Talked offline, decided on finding a better name for `variables` that expresses it's dual purpose as 1. variables that were referenced in the query 2. variable-to-type mapping